classes/class-p4-control-panel.php

```diff
@@ -0,0 +1,175 @@
+
+			add_action( 'wp_ajax_check_cache',          array( $this, 'check_cache' ) );
+			add_action( 'wp_ajax_check_en',             array( $this, 'check_en' ) );
+
+			add_action( 'admin_enqueue_scripts',        array( $this, 'enqueue_admin_assets' ) );
+		}
+
+		/**
+		 * .
+		 */
+		public function add_dashboard_widgets() {
+			wp_add_dashboard_widget(
+				'planet4_control_panel',
+				__( 'Planet 4 Control Panel', 'planet4-master-theme' ),
+				array( $this, 'add_items' )
+			);
+		}
+
+		/**
+		 * Outputs the contents of the dashboard widget.
+		 */
+		public function add_items() {
+			wp_nonce_field( 'cp-action' );
+
+			$this->add_item( [
+				'title'    => __( 'Cache', 'planet4-master-theme' ),
+				'subitems' => [
+					[
+						'title'  => __( 'Check Object Cache', 'planet4-master-theme' ),
+						'action' => 'check_cache',
+					],
+				],
+			] );
+
+			$this->add_item( [
+				'title'    => __( 'Engaging Networks', 'planet4-master-theme' ),
+				'subitems' => [
+					[
+						'title'  => __( 'Check Engaging Networks', 'planet4-master-theme' ),
+						'action' => 'check_en',
+					],
+				],
+			] );
+		}
+
+		/**
+		 * Adds a new item in the Control Panel and all of its subitems.
+		 *
+		 * @param array $data Associative array with all the data needed to add a new item in the Control Panel.
+		 */
+		public function add_item( $data ) {
+			echo '<div class="cp-item">
+					<div class="welcome-panel"><span><strong>' . esc_html( $data['title'] ) . '</strong></span>';
+			foreach ( $data['subitems'] as $subitem ) {
+				echo '<div>
+						<a href="#" class="btn btn-cp-action btn-' . esc_attr( $subitem['action'] ) . '-async" data-action="' . esc_attr( $subitem['action'] ) . '">' . esc_html( $subitem['title'] ) . '</a>
+						<span class="cp-subitem-response"></span>
+					</div>';
+			}
+			echo '</div>
+				</div>';
+		}
+
+		/**
+		 * Adds a check cache button to check connectivity to the Redis server.
+		 */
+		public function check_cache() {
+			if ( wp_doing_ajax() ) {
+				if ( ! current_user_can( 'manage_options' ) ) {
+					return;
+				}
+
+				if ( wp_verify_nonce( $_GET['_wpnonce'], 'cp-action' )  // CSRF protection.
+				     && isset( $_GET['cp-action'] )
+				     && 'check_cache' === $_GET['cp-action']
+				) {
+					$messages = [];
+					$info     = wp_redis_get_info();
+
+					if ( $info instanceof WP_Error ) {
+						if ( $info->errors['wp-redis'] ) {
+							foreach ( $info->errors['wp-redis'] as $index => $error ) {
+								$messages[] = $error;
+							}
+							$class = 'cp-error';
+						}
+					} elseif ( 'connected' === $info['status'] ) {
+						$messages[] = __( 'Planet 4 is connected to Redis.', 'planet4-master-theme' );
+						$class      = 'cp-success';
+					}
+
+					if ( $messages ) {
+						echo wp_json_encode( [ $messages, $class ] );
+					}
+				}
+				wp_die();
+			}
+		}
+
+		/**
+		 * Adds a check cache button to check the ENS API.
+		 */
+		public function check_en() {
```

Comment:
I'd rename this to `check_engaging_networks()` to avoid any ambiguity, this reads to me like you're checking an english translation.